### PR TITLE
Add commatrix tests

### PR DESCRIPTION
**Pull request by liornoy**
Coverage tests,

Please only consider the second commit: "Refactor commatrix tests"
The first one is kept for reference and will be squashed before merge.

Also - the node-comm-lib will eventually be moved to a more suitable place
e.g. openshift/library-go

### CODE REVIEW

File: test/extended/networking/commatrix.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	"github.com/liornoy/node-comm-lib/commatrix"
```
Comment (yanjoseph) [inline test/extended/networking/commatrix.go:13]: any discussion made WRT a suitable host for the code ?

Comment (liornoy) [reply test/extended/networking/commatrix.go:13]: we're considering moving it to openshift/library-go.
this is pointing to my own repo temporarily just to see first that the library does what we want it to do.

Comment (liornoy) [reply test/extended/networking/commatrix.go:13]: or even under openshift/oc

File: test/extended/networking/commatrix.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer tcpfile.Close()
+
+		udpfile, err := os.OpenFile(path.Join(artifactsDir, "raw-ss-udp"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer tcpfile.Close()
+
+		nodesComDeatils := []types.ComDetails{}
+		for _, n := range nodesList.Items {
+			cds, err := ssutil.CreateComDetailsFromNode(oc, &n, tcpfile, udpfile)
```
Comment (sabinaaledort) [inline test/extended/networking/commatrix.go:57]: To be more clear on what this is doing let's consider changing to ss.ToComDetails or ss.ConvertToComDetails

File: test/extended/networking/commatrix.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			g.Fail("must set the KUBECONFIG environment variable")
+		}
+
+		comMatrix, err := commatrix.New(kubeconfig, "", commatrix.AWS)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		cs, err := clientutil.New(kubeconfig)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("fetching open ports on the cluster")
```
Comment (sabinaaledort) [inline test/extended/networking/commatrix.go:43]: Let's be more clear here, this is fetching open ports on the nodes with ss

File: test/extended/util/ss/ss.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	ssOutFilteredTCP := filterStrings(tcpSSFilterFn, splitByLines(ssOutTCP))
+	ssOutFilteredUDP := filterStrings(udpSSFilterFn, splitByLines(ssOutUDP))
+
+	_, err = tcpfile.Write([]byte(fmt.Sprintf("node: %s\n%s", node.Name, strings.Join(ssOutFilteredTCP, "\n"))))
+	if err != nil {
+		return nil, fmt.Errorf("failed writing to file: %s", err)
+	}
+	_, err = tcpfile.Write([]byte(fmt.Sprintf("node: %s\n%s", node.Name, strings.Join(ssOutFilteredUDP, "\n"))))
```
Comment (sabinaaledort) [inline test/extended/util/ss/ss.go:57]: Both UDP and TCP are printed to TCP file

**Comment by liornoy** [comment]
closing this in favor of the updated test https://github.com/openshift/origin/pull/28706

**Pull request closed without merging**